src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -106,6 +106,36 @@ internal sealed partial class ObjectReader : IDisposable
             return new ObjectReader(stream, leaveOpen, cancellationToken);
         }
 
+        public static ObjectReader GetReader(
+            Stream stream,
+            bool leaveOpen,
+            CancellationToken cancellationToken)
+        {
+            var b = stream.ReadByte();
+            if (b == -1)
+            {
+                throw new EndOfStreamException();
+            }
+
+            if (b != VersionByte1)
+            {
+                throw ExceptionUtilities.UnexpectedValue(b);
```

Comment:
i think this is breaking the pervious contract we had here, namely that if hte version changes we get `null` back.  callers are written expecting null to mean that the format changed and coudl no longer be read, and thus they should recompute the data.

Reply:
Not breaking since this is a new method.

Reply:
The readers that can handle a different version should keep calling TryGetReader.

Reply:
please doc the semantics of this method then :)

Reply:
Sure.